src/storage/s3.rs

```diff
@@ -56,19 +29,20 @@ impl S3Backend {
     pub(super) fn new(metrics: Arc<Metrics>, config: &Config) -> Result<Self, Error> {
         let runtime = Runtime::new()?;
 
-        // Connect to S3
-        let client = S3Client::new_with(
-            rusoto_core::request::HttpClient::new()?,
-            DefaultCredentialsProvider::new()?,
-            config
-                .s3_endpoint
-                .as_deref()
-                .map(|endpoint| Region::Custom {
-                    name: config.s3_region.name().to_string(),
-                    endpoint: endpoint.to_string(),
-                })
-                .unwrap_or_else(|| config.s3_region.clone()),
-        );
+        let shared_config = runtime.block_on(aws_config::load_from_env());
+        let mut config_builder = aws_sdk_s3::config::Builder::from(&shared_config)
+            .retry_config(RetryConfig::new().with_max_attempts(5))
```

Comment:
We had 3 max attempts before - can we keep using that?
```suggestion
            .retry_config(RetryConfig::new().with_max_attempts(3))
```